wandb/sdk/artifacts/storage_handlers/http_handler.py

```diff
@@ -45,7 +46,12 @@ def load_path(
         if hit:
             return path
 
-        response = self._session.get(manifest_entry.ref, stream=True)
+        response = self._session.get(
+            manifest_entry.ref,
+            stream=True,
+            cookies=_thread_local_api_settings.cookies,
+            headers=_thread_local_api_settings.headers,
```

Comment:
This can talk to any server. You're sending W&B cookies & headers to that server.
```
artifact = wandb.Artifact("my_artifact", type="my_type")
artifact.add_reference("https://some.server")
artifact.save()
```